crates/polars-lazy/src/scan/file_list_reader.rs

```diff
@@ -104,19 +205,9 @@ pub trait LazyFileListReader: Clone {
         true
     }
 
-    /// Path of the scanned file.
-    /// It can be potentially a glob pattern.
-    fn path(&self) -> &Path;
```

Comment:
Removed the `path` function from the trait - we just use a length-1 slice in `paths` for single paths